pkg/scheduler/scheduler.go

```diff
@@ -97,46 +97,109 @@ type Configurator interface {
 }
 
 // Config is an implementation of the Scheduler's configured input data.
-// TODO over time we should make this struct a hidden implementation detail of the scheduler.
 type Config struct {
```

Comment:
Do we know there are no consumers driving the scheduler via this? /cc @aveshagarwal

Reply:
Is a config struct with private members like this consistent with how other components are driven?

Reply:
For `kube-apiserver`, `kubelet` and `kube-controller-manager`, all of them expose the config's members. Now I doubt if the `TODO` is reasonable or there are special purposes of it.

Reply:
@liggit Please give some suggestions on if the `TODO` is necessary? Thanks!
:smiley:

Reply:
I would probably remove the TODO and wait to move scheduler in this direction until it's clear all the components are doing so

Reply:
Agreed. I will just remove the `TODO` then.